scripts/utilities/datarun.py

```diff
@@ -0,0 +1,360 @@
+                
+        
+
+    def __str__(self):
+        '''A very generic __str__ that just gets everythiing that's not a built-in
+        or a method.'''
+        # get the attributes that matter
+        # from stack-exchange
+        # https://stackoverflow.com/questions/11637293/iterate-over-object-attributes-in-python
+        attrs = [a for a in dir(self) if not a.startswith('__') and not callable(getattr(self,a))]
+        s = []
+        for attr in attrs:
+            s.append(attr + ': ' + str(self.__dict__[attr]))
+        return '\n'.join(s)
+    
+    
+    def setup(self):
+        '''Create the experiment directory and the .par file. Create the .csp input 
+        using genesis if neccesary. Deal with the file path to real_data.
+        Get the args for the call to rescal.'''
+
+        # some files and directories should alreay exist
+        self.check_paths()
+        
+        # make the experiment directory
+        if not os.path.exists(self.experiment_directory):
+            os.mkdir(self.experiment_directory)
+
+        # deal with .csp files
+        # if 'premade_csp' overwrite 'Csp_file'
+        if 'premade_csp' in self.parameters.keys():
+            # put the .csp name into the .par file as Csp_file
+            design.set_parameters({'Csp_file' : self.parameters['premade_csp']})
+        else:
+            # if 'Csp_file' just a file name, put it into self.experiment_directory
+            if 'Csp_file' in self.parameters.keys():
+                # if it's an absolute path, leave it alone, otherwise
+                # take the file name and put it into the experiment directory
+                if not os.path.isabs(self.parameters['Csp_file']):
+                    self.parameters['Csp_file'] = os.path.join(self.experiment_directory,
+                                                               os.path.basename(self.parameters['Csp_file']))
+            else:
+                # give 'Csp_file' a generic name
+                self.parameters['Csp_file'] = os.path.join(self.experiment_directory, 'Cells.csp')
+
+        # set output directory
+        # TODO allow users to specify their own output directory
+        self.parameters['Output_directory'] = os.path.join(self.experiment_directory, 'out')
+                
+        # create a run with the given parameters
+        design = rescal_utilities.Design_a_run()
+        design.set_parameters(self.parameters)
+
+        # find file paths that are in the real data folder and fix them
+        # using self.real_data_directory 
+        design.parameters.set_real_data_path(self.real_data_directory)
+            
+        # write the par file
+        design.parameters.write(self.par)
+
+        # run genesis if neccesary
+        if 'premade_csp' not in self.parameters.keys():
+            # run genesis and put the output where specified by 'Csp_file' parameter
+            p = subprocess.Popen([self.genesis_executable, '-f',  self.par, '-s', str(2000)],
```

Comment:
Consider `subprocess.check_output`

Reply:
I don't need or want the output in this case, so I'll use subprocess.check_call.

Reply:
Sounds good. If you update and push to this branch, the PR will update automatically.